### PR TITLE
Fix node env

### PR DESCRIPTION
**Pull request by WaldoJeffers**
This fixes https://github.com/styleguidist/react-styleguidist/issues/677

- Add a new environment variable called `STYLEGUIDIST_ENV` to check if we're in dev server or static build mode
- Allow the user to pass `NODE_ENV` to Webpack's config, to determine if React will use its production or development build

### CODE REVIEW

File: docs/Cookbook.md [hunk 1]
```diff
@@ -260,6 +261,32 @@ module.exports = {
...
+};
+```
+If you use code similar to this, you might encounter errors such as `Cannot read property 'isRequired' of undefined`.
+
+To avoid this, you need to tell React Styleguidist to use React's development build. To do this, simply set the `NODE_ENV` variable to `development` in your npm script.
+
+```json
+{
+  "scripts": {
+    "build": "NODE_ENV=development react-styleguidist build"
```
Comment (sapegin) [inline docs/Cookbook.md]: Would be nice to use cross-env here.

Comment (WaldoJeffers) [reply docs/Cookbook.md]: Good point

**Comment by WaldoJeffers** [comment]
@sapegin Wait, please **do not merge**. When doing a more thorough test, I am now unavailable to replicate the results I was obtaining using `dangerouslyUpdateWebpackConfig` and definePlugin manual override. Will link into that.

**Comment by WaldoJeffers** [comment]
@sapegin actually nevermind. I was seeing weird results because of `npm link`. Has anyone else encountered issues with that? Had to publish my branch on a scoped npm package, to make sure everything was running smoothly.

**Comment by sapegin** [comment]
Merged, thanks!

**Pull request merged**